Switch to using a specific version of CentOS Stream

jovial opened the pull request:
This is to workaround the following breakage:
```
TASK [MichaelRigart.interfaces : RedHat | ensure network service is started and enabled] ***
Tuesday 21 March 2023  09:43:32 +0000 (0:00:00.347)       0:02:20.016 ********* 
fatal: [controller0]: FAILED! => changed=false 
  msg: |-
    Unable to start service network: Job for network.service failed because the control process exited with error code.
    See "systemctl status network.service" and "journalctl -xe" for details.
```

jovial committed:
Switch to using a specific version of CentOS Stream

This is to workaround the following breakage:
```
TASK [MichaelRigart.interfaces : RedHat | ensure network service is started and enabled] ***
Tuesday 21 March 2023  09:43:32 +0000 (0:00:00.347)       0:02:20.016 ********* 
fatal: [controller0]: FAILED! => changed=false 
  msg: |-
    Unable to start service network: Job for network.service failed because the control process exited with error code.
    See "systemctl status network.service" and "journalctl -xe" for details.
```

jovial commented:
~~Following existing convention set by Rocky: https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/wallaby/.github/workflows/stackhpc-pull-request.yml#L94. Will switch to UUID if people think that is more robust. The only issue is that UUIDs aren't human friendly.~~

jovial commented on terraform/aio/vm.tf:
```diff
@@ -13,7 +13,7 @@ variable "aio_vm_name" {
 variable "aio_vm_image" {
   type    = string
-  default = "CentOS-stream8"
+  default = "CentOS-stream8.2023-03-20"
```
Dammit, this one blew up too:
```
RUNNING HANDLER [MichaelRigart.interfaces : Bounce network devices] ************
fatal: [controller0]: FAILED! => {"changed": true, "cmd": ["nohup", "bash", "-c", " returncode=0  ifdown breth1;  ifdown eth0;  if ! ifup eth0; then\n  echo \"Failed to bring up interface eth0\";\n  returncode=1\nfi;  if ! ifup breth1; then\n  echo \"Failed to bring up interface breth1\";\n  returncode=1\nfi; exit $returncode"], "delta": "0:00:04.926768", "end": "2023-03-21 14:08:57.695881", "msg": "non-zero return code", "rc": 1, "start": "2023-03-21 14:08:52.769113", "stderr": "nohup: ignoring input\nWARN      : [ifdown] You are using 'ifdown' script provided by 'network-scripts', which are now deprecated.\nWARN      : [ifdown] 'network-scripts' will be removed in one of the next major releases of RHEL.\nWARN      : [ifdown] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well.\nWARN      : [ifdown] You are using 'ifdown' script provided by 'network-scripts', which are now deprecated.\nWARN      : [ifdown] 'network-scripts' will be removed in one of the next major releases of RHEL.\nWARN      : [ifdown] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well.\nWARN      : [ifup] You are using 'ifup' script provided by 'network-scripts', which are now deprecated.\nWARN      : [ifup] 'network-scripts' will be removed in one of the next major releases of RHEL.\nWARN      : [ifup] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well.\nWARN      : [ifup] You are using 'ifup' script provided by 'network-scripts', which are now deprecated.\nWARN      : [ifup] 'network-scripts' will be removed in one of the next major releases of RHEL.\nWARN      : [ifup] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well.", "stderr_lines": ["nohup: ignoring input", "WARN      : [ifdown] You are using 'ifdown' script provided by 'network-scripts', which are now deprecated.", "WARN      : [ifdown] 'network-scripts' will be removed in one of the next major releases of RHEL.", "WARN      : [ifdown] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well.", "WARN      : [ifdown] You are using 'ifdown' script provided by 'network-scripts', which are now deprecated.", "WARN      : [ifdown] 'network-scripts' will be removed in one of the next major releases of RHEL.", "WARN      : [ifdown] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well.", "WARN      : [ifup] You are using 'ifup' script provided by 'network-scripts', which are now deprecated.", "WARN      : [ifup] 'network-scripts' will be removed in one of the next major releases of RHEL.", "WARN      : [ifup] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well.", "WARN      : [ifup] You are using 'ifup' script provided by 'network-scripts', which are now deprecated.", "WARN      : [ifup] 'network-scripts' will be removed in one of the next major releases of RHEL.", "WARN      : [ifup] It is advised to switch to 'NetworkManager' instead - it provides 'ifup/ifdown' scripts as well."], "stdout": "ERROR     : [/etc/sysconfig/network-scripts/ifup-eth] Device eth0 does not seem to be present, delaying initialization.\nFailed to bring up interface eth0", "stdout_lines": ["ERROR     : [/etc/sysconfig/network-scripts/ifup-eth] Device eth0 does not seem to be present, delaying initialization.", "Failed to bring up interface eth0"]}
```

jovial replied:
```suggestion
  default = "bb8c0a34-533f-42fb-a49b-3461e677f3f6"
```

jovial committed:
Switch to UUID

Retrieved this version from: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/4449862721/jobs/7814765240#step:8:56

jovial commented on terraform/aio/vm.tf:
```diff
@@ -13,7 +13,7 @@ variable "aio_vm_name" {
 variable "aio_vm_image" {
   type    = string
-  default = "CentOS-stream8"
+  default = "bb8c0a34-533f-42fb-a49b-3461e677f3f6"
```
```suggestion
  default = "7efda0b1-e296-447f-896e-c066d0af5c53"
```

From: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/4450001402/jobs/7815038031#step:8:56

jovial committed:
Switch to UUID from known good xena job

From: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/4450001402/jobs/7815038031#step:8:56

jovial committed:
Try updating interface naming scheme

markgoddard commented on terraform/aio/vm.tf:
```diff
@@ -13,12 +13,12 @@ variable "aio_vm_name" {
 variable "aio_vm_interface" {
   type = string
-  default = "eth0"
+  default = "ens3"
```
hmm, why would it need to change if we're keeping the same image?

jovial replied:
That was also my thought. Let me look into a bit closer.

jovial replied:
Crikey, the default comes from here: 

https://github.com/stackhpc/stackhpc-kayobe-config/blob/stackhpc/wallaby/.github/workflows/stackhpc-all-in-one.yml#L25

and not the terraform :see_no_evil:

jovial replied:
Marking as resolved as I have removed this change. I was curious to see if that would fix the issue with new image.

jovial commented on terraform/aio/vm.tf:
```diff
@@ -13,12 +13,12 @@ variable "aio_vm_name" {
 variable "aio_vm_interface" {
   type = string
-  default = "eth0"
+  default = "ens3"
```
```suggestion
  default = "eth0"
```

jovial commented on terraform/aio/vm.tf:
```diff
@@ -13,12 +13,12 @@ variable "aio_vm_name" {
 variable "aio_vm_image" {
   type    = string
-  default = "CentOS-stream8"
+  default = "7efda0b1-e296-447f-896e-c066d0af5c53"
```
```suggestion
  default = "CentOS-stream8"
```

jovial committed:
Update image in correct location

jovial committed:
Revert previous changes

jovial commented on .github/workflows/stackhpc-all-in-one.yml:
```diff
@@ -22,7 +22,7 @@ on:
         description: Image for the all-in-one VM
         type: string
-        default: CentOS-stream8
+        default: 7efda0b1-e296-447f-896e-c066d0af5c53
```
```suggestion
        default: bb8c0a34-533f-42fb-a49b-3461e677f3f6
```
Copied from the wrong job. Using this one: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/4450001402/jobs/7815037603#step:8:56

jovial committed:
Use last known good image

From: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/4450001402/jobs/7815037603#step:8:56

jovial committed:
Autodetect uuids

jovial commented on terraform/aio/vm.tf:
```diff
@@ -33,9 +33,14 @@ variable "aio_vm_subnet" {
 }
 
+locals {
+  image_is_uuid = length(regexall('^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$', aio_vm_image)) > 0
```
```suggestion
  image_is_uuid = length(regexall('^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$', var.aio_vm_image)) > 0
```

jovial committed:
Fixup for previous commit

jovial commented on terraform/aio/vm.tf:
```diff
@@ -33,9 +33,14 @@ variable "aio_vm_subnet" {
 }
 
+locals {
+  image_is_uuid = length(regexall('^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$', var.aio_vm_image)) > 0
```
```suggestion
  image_is_uuid = length(regexall("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$", var.aio_vm_image)) > 0
```

jovial committed:
Use single quotes for strings

jovial commented on terraform/aio/vm.tf:
```diff
@@ -52,7 +57,7 @@ resource "openstack_compute_instance_v2" "kayobe-aio" {
 
   block_device {
-    uuid                  = data.openstack_images_image_v2.image.id
+    uuid                  = local.image_is_uuid ? var.aio_vm_image: data.openstack_images_image_v2.image.id
```
```suggestion
    uuid                  = local.image_is_uuid ? var.aio_vm_image: data.openstack_images_image_v2.image[0].id
```

jovial committed:
Resource using count must access via index

markgoddard commented on terraform/aio/vm.tf:
```diff
@@ -33,9 +33,14 @@ variable "aio_vm_subnet" {
 }
 
+locals {
+  image_is_uuid = length(regexall("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$", var.aio_vm_image)) > 0
```
:(
